// .ignored/vue-element-plus-x/dist/es/index.js
import Attachments from "./Attachments/index.js";
import Bubble from "./Bubble/index.js";
import BubbleList from "./BubbleList/index.js";
import Conversations from "./Conversations/index.js";
import EditorInput from "./EditorInput/index.js";
import EditorSender from "./EditorSender/index.js";
import { F as FilesCard } from "../index-BvufZeL0.js";
import MentionSender from "./MentionSender/index.js";
import Prompts from "./Prompts/index.js";
import Sender from "./Sender/index.js";
import Thinking from "./Thinking/index.js";
import ThoughtChain from "./ThoughtChain/index.js";
import { T as Typewriter } from "../index-CD6jtNKK.js";
import Welcome from "./Welcome/index.js";
import { u } from "../usePrism-4BCCicak.js";
import { useRecord } from "./hooks/useRecord.js";
import { XRequest, useSend } from "./hooks/useSend.js";
import { useXStream } from "./hooks/useXStream.js";
const ElementPlusX = {
  install(app) {
    app.component("Attachments", Attachments);
    app.component("Bubble", Bubble);
    app.component("BubbleList", BubbleList);
    app.component("Conversations", Conversations);
    app.component("EditorInput", EditorInput);
    app.component("EditorSender", EditorSender);
    app.component("FilesCard", FilesCard);
    app.component("MentionSender", MentionSender);
    app.component("Prompts", Prompts);
    app.component("Sender", Sender);
    app.component("Thinking", Thinking);
    app.component("ThoughtChain", ThoughtChain);
    app.component("Typewriter", Typewriter);
    app.component("Welcome", Welcome);
  }
};
export {
  Attachments,
  Bubble,
  BubbleList,
  Conversations,
  EditorInput,
  EditorSender,
  FilesCard,
  MentionSender,
  Prompts,
  Sender,
  Thinking,
  ThoughtChain,
  Typewriter,
  Welcome,
  XRequest,
  ElementPlusX as default,
  u as usePrism,
  useRecord,
  useSend,
  useXStream
};
